shotover-proxy/src/transforms/mod.rs

```diff
@@ -521,6 +584,27 @@ pub trait Transform: Send {
     /// Wrapping it in an [`Arc<Mutex<_>>`](std::sync::Mutex) would make it a global count of all messages seen by this transform.
     async fn transform<'a>(&'a mut self, message_wrapper: Wrapper<'a>) -> ChainResponse;
 
+    /// This method should be should be implemented by your transform if it is required to process pushed messages (typically events
+    /// or messages that your source is subscribed to. The wrapper object contains the queries/frames
+    /// in a `[Vec<Message]`(crate::message::Message).
+    ///
+    /// This transform method is not the same request/response model as the other transform method.
+    /// This method processes one pushed message before sending it in reverse on the chain back to the source.
+    ///
+    /// You can modify the messages in the wrapper struct to achieve your own designs. Your transform can
+    /// also modify the response from `message_wrapper.call_next_transform_rev` if it needs to. As long as the message
+    /// carries on through the chain, it will function correctly.
+    ///
+    /// ## Invariants
+    /// * _Non-terminating_ - Your `transform_rev` method should not be terminating as the messages should get passed back to the source, where they will terminate.
+    /// * _Message count_ - you can modify pushed messages but you should not remove or add them.
```

Comment:
What about message ordering?

Reply:
Also... Is there actually a requirement for message count to remain the same here?
Its not like the protocol has to match requests up with responses here

Reply:
I think we can allow transform authors to add or remove messages? Given it's not expecting request/reponse pairs?